### PR TITLE
Dialog: Better breakpoint management and refinement

### LINKED ISSUE #371: Dialog: the dialog is not centered correctly, and is too thin, at certain screen sizes
The current version of the Design System calls for 8 or 12 columns for the dialog, but doesn't outline any steps in between for varying screens.

A fix would be to add proper breakpoints for the dialog across a host of screen sizes, and adjust the number of columns used.

The use case below is for an offer dialog and the thin width for a desktop-like screen is poor.

### Example: Home page offer

**dWeb - wide - 1639**
dweb dialog

**dWeb - thin - 1024**
dweb dialog

### PR DESCRIPTION
**Pull request by seangates**
## Description
- sets a default max-width
- creates new breakpoints at 600px, 768px and 1024px
- adds better handling for when the grid changes to calculate gutters and margins

## Context
The crux of this change is to make it better for the dialog on the widths _in between_, as much as it's handling the widths on the lower end, and the higher end.

To that point, the newer guidance from our Design System partners has helped a lot.

## References
Fixes #371

### Before

**iPhone 6/7/8 portrait**

image

**iPad portrait**

image

**iPad landscape**

image

### After

**Galaxy S5 portrait**

image

**iPhone 6/7/8 portrait**

image

**iPhone 6/7/8 landscape**

image

**iPad portrait**

image

**iPad Landscape**

image

**Desktop**

image

### CODE REVIEW

File: dist/dialog/ds4/dialog.css [hunk 1]
```diff
@@ -162,11 +161,24 @@
...
+  .dialog__window:not(.dialog__window--full) {
+    width: calc(88% - 16px);
+    min-width: calc(88% - 16px);
+    max-width: calc(88% - 16px);
+  }
+}
+@media (min-width: 769px) and (max-width: 1024px) {
+  .dialog__window:not(.dialog__window--full) {
+    width: calc(75% - 32px);
+    max-width: 616px;
```
Comment (RajaRamu) [inline dist/dialog/ds4/dialog.css]: i think this is not needed since its set default(L18)

Comment (seangates) [reply dist/dialog/ds4/dialog.css]: Yes, it’s needed. Without it the higher order selector overrides the width settings and it becomes too wide.

Please try and remove it and test it to confirm.

Comment (RajaRamu) [reply dist/dialog/ds4/dialog.css]: and, all the desktop dialog would be small compare to the big screens since we its restricted to max-width 616px. I think 616px on bigger screen would look like small modal. max-width can be set in percentage of viewport. thought?
 and, pages support following breakpoints,
@breakpoint-smallest: 480px;
@breakpoint-smaller:  600px;
@breakpoint-small:    768px;
@breakpoint-medium:   960px;
@breakpoint-large:    1024px;
@breakpoint-larger:   1280px;
@breakpoint-wide-ads: 1350px;
@breakpoint-largest:  1420px;

Comment (seangates) [reply dist/dialog/ds4/dialog.css]: Yeah, and those current breakpoints don't apply to a dialog. It has it's own breakpoints and own set of rules.

File: docs/src/less/highlight.less [hunk 2]
```diff
@@ -19,3 +19,15 @@ figure.highlight {
         white-space: pre-wrap;
     }
 }
+
+@media (max-width: 600px) {
+    a,
```
Comment (ianmcburnie) [inline docs/src/less/highlight.less]: I don't think we should be adding global rules inside highlight.less. Seems like the wrong place for these. Same goes for the the `demo` selector. I'd actually prefer not have any of this as part of the PR if possible, as it seems unrelated. I'm guessing they help for the dialog examples, but I'm not sure what effect it's going to have in other places.

Comment (seangates) [reply docs/src/less/highlight.less]: So ... this was for overflows that were breaking the page on smaller screens, and it's specific to the highlighting of the code.

And, I could not test the dialog properly without these changes in place.

Comment (ianmcburnie) [reply docs/src/less/highlight.less]: Ok, but they shouldn't live in highlight.less. That file is specific to the Pygments/Jekyll stuff.

Comment (ianmcburnie) [reply docs/src/less/highlight.less]: These rules are far reaching, so if there's any way to make it more specific to dialog, that would be good. Not sure we really want all h2s and anchors to have break word.

Comment (seangates) [reply docs/src/less/highlight.less]: Removed.
